src/coordination/coordinator_state_manager.cpp

```diff
@@ -73,29 +90,36 @@ auto CoordinatorStateManager::save_config(cluster_config const &config) -> void
   ptr<buffer> buf = config.serialize();
   cluster_config_ = cluster_config::deserialize(*buf);
   logger_.Log(nuraft_log_level::TRACE, "Saving cluster config to RocksDb");
-  auto const servers_vec =
-      ranges::views::transform(
-          cluster_config_->get_servers(),
-          [this](auto const &server) {
-            logger_.Log(nuraft_log_level::TRACE,
-                        fmt::format("Creating cluster config with id: {}, endpoint: {} and aux data: {} to disk.",
-                                    static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux()));
-            return std::tuple{static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux()};
-          }) |
-      ranges::to<std::vector>();
-  kv_store_.Put(kServersKey, nlohmann::json(servers_vec).dump());
+  auto json = SerializeClusterConfig(config);
+  MG_ASSERT(state_manager_durability_.Put(kClusterConfigKey, json.dump()), "Failed to save servers to disk");
 }
 
 auto CoordinatorStateManager::save_state(srv_state const &state) -> void {
-  // TODO(antoniofilipovic): Implement storing of server state to disk. For now
-  // as server state is just term and voted_for, we don't have to store it
   logger_.Log(nuraft_log_level::TRACE, "Saving server state in coordinator state manager.");
+
+  auto const server_state_json = nlohmann::json{{kTerm, state.get_term()},
+                                                {kVotedFor, state.get_voted_for()},
+                                                {kElectionTimer, state.is_election_timer_allowed()}};
+  MG_ASSERT(state_manager_durability_.Put(kServerStateKey, server_state_json.dump()),
```

Comment:
Not sure if we should assert false here. Nothing will break immediately, I would just log instead of crashing Memgraph. We cannot influence whether RocksDB will crash or not

Reply:
I would like to sync on this one with you